### ISSUE TITLE
FreeFleet and This adapter publish in the same topic /fleet_states

### ISSUE THREAD

**Issue by TheConstructAi** [open-rmf/fleet_adapter_tb4#2]
Here you can see that using this TB4 adapter with a freefleet enabled robot, seems to work partially, but has this strange flickering. This is because it seems that we have both the adapter and the freelfeet publishing in `/fleet_states` topic.

This makes navigation a bit unstable.

Is there a way to fix this, or is there something that in this new adapters is done differently with this easFullcontrol?

### Required information:

- Operating system and version: Ubuntu 22
- OpenRMF installation type: Source with the repos used in the roscon 2022 demos
- OpenRMF version or commit hash

### Description of the bug
With No translation in the freefleetserver
![ezgif-2-08ff1d9c4b](https://user-images.githubusercontent.com/80715513/203982683-f4aac31b-c03a-4298-8191-27c2f3df4579.gif)

With adjusted translations to minimise this difference:
![smallmovement](https://user-images.githubusercontent.com/80715513/203982692-20f18963-0abb-4e48-a792-1af8d6064d98.gif)

##

Is there something in the inner workings now that we are unaware of? Thanks and hope to hear form you soon

### 

Here are the connections to the `fleet_states` topic:

```
os2 topic info /fleet_states --verbose
Type: rmf_fleet_msgs/msg/FleetState

Publisher count: 2

Node name: barista_fleet_server_node
Node namespace: /
Topic type: rmf_fleet_msgs/msg/FleetState
Endpoint type: PUBLISHER
GID: 01.10.19.77.8a.52.65.c8.28.59.4c.c5.00.00.15.03.00.00.00.00.00.00.00.00
QoS profile:
  Reliability: RELIABLE
  History (Depth): UNKNOWN
  Durability: VOLATILE
  Lifespan: Infinite
  Deadline: Infinite
  Liveliness: AUTOMATIC
  Liveliness lease duration: Infinite

Node name: fleet_adapter_barista_ex1
Node namespace: /
Topic type: rmf_fleet_msgs/msg/FleetState
Endpoint type: PUBLISHER
GID: 01.10.78.65.69.5b.04.e3.79.74.c9.da.00.00.32.03.00.00.00.00.00.00.00.00
QoS profile:
  Reliability: RELIABLE
  History (Depth): UNKNOWN
  Durability: VOLATILE
  Lifespan: Infinite
  Deadline: Infinite
  Liveliness: AUTOMATIC
  Liveliness lease duration: Infinite

Subscription count: 4

Node name: bartender_dispenser_1
Node namespace: /
Topic type: rmf_fleet_msgs/msg/FleetState
Endpoint type: SUBSCRIPTION
GID: 01.10.ac.86.c3.4f.b9.9e.c1.3f.18.75.00.00.3e.04.00.00.00.00.00.00.00.00
QoS profile:
  Reliability: RELIABLE
  History (Depth): UNKNOWN
  Durability: VOLATILE
  Lifespan: Infinite
  Deadline: Infinite
  Liveliness: AUTOMATIC
  Liveliness lease duration: Infinite

Node name: table_ingestor_1_node
Node namespace: /
Topic type: rmf_fleet_msgs/msg/FleetState
Endpoint type: SUBSCRIPTION
GID: 01.10.ac.86.c3.4f.b9.9e.c1.3f.18.75.00.00.52.04.00.00.00.00.00.00.00.00
QoS profile:
  Reliability: RELIABLE
  History (Depth): UNKNOWN
  Durability: VOLATILE
  Lifespan: Infinite
  Deadline: Infinite
  Liveliness: AUTOMATIC
  Liveliness lease duration: Infinite

Node name: fleet_states_visualizer
Node namespace: /
Topic type: rmf_fleet_msgs/msg/FleetState
Endpoint type: SUBSCRIPTION
GID: 01.10.ce.7b.42.78.b4.6b.3c.be.30.43.00.00.17.04.00.00.00.00.00.00.00.00
QoS profile:
  Reliability: RELIABLE
  History (Depth): UNKNOWN
  Durability: VOLATILE
  Lifespan: Infinite
  Deadline: Infinite
  Liveliness: AUTOMATIC
  Liveliness lease duration: Infinite

Node name: simple_api_server
Node namespace: /
Topic type: rmf_fleet_msgs/msg/FleetState
Endpoint type: SUBSCRIPTION
GID: 01.10.d7.4b.d7.d0.1a.a4.6b.c7.30.ee.00.00.18.04.00.00.00.00.00.00.00.00
QoS profile:
  Reliability: RELIABLE
  History (Depth): UNKNOWN
  Durability: VOLATILE
  Lifespan: Infinite
  Deadline: Infinite
  Liveliness: AUTOMATIC
  Liveliness lease duration: Infinite
```

When setting the free fleet to publish elsewhere from `/fleet_states` , the planner works , so the freefleet publishing is a bit of a mystery here. So.. Is this adapter using the freefleet only as a bridge to navigation? 

Looking into the TB4 adapter, the robot state is retrieved from the `/my_robot_name/odom` , which is strange because that is not a good source of information on the real position of the robot, that's why we need navigation to recalculate the pose unless the odometry system in the robot is bulletproof. The example in my system is:

```
ros2 topic info /barista_1/odom --verbose

Publisher count: 1

Node name: diff_drive

Subscription count: 2

Node name: controller_server
Node name: fleet_adapter_barista_ex1
```

So it seems that freefleet is just a navigation goal bridge of somekind?

Is this `fleet_states` only used by the visualization and Web systems? Months ago, freefleet and the adapters worked flawlessly, but now this fleet_states function is a bit strange.

Hope I made the issue clear.

**Comment by Yadunund** [comment]
I think there is some misunderstanding with the different adapters.

Free Fleet is something that should not be run alongside this adapter or any of our other adapters EXCEPT our [legacy full_control](https://github.com/open-rmf/rmf_ros2/blob/main/rmf_fleet_adapter/launch/fleet_adapter.launch.xml) adapter.

If you look at the implementation of that legacy adapter, it subscribes to /fleet_states and then adds robots to the fleet to be managed by Open-RMF. Hence, it is designed to run together with free_fleet_server which publishes /fleet_states.

The adapter here and others do not rely on this mechanism. Instead, they talk to the robots over the robot's API (in this case ROS 2) to directly get information on the robot's state and issue commands to the robot.

TLDR
You should run this adapter independently and without launching any free_fleet services.

**Comment by Yadunund** [comment]
Closing this ticket as I assume the issue is resolved. Feel free to reopen otherwise.
